.github/workflows/publish-docs.yaml

```diff
@@ -26,25 +38,47 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
 
-      - name: Setup Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: 3.x
-          cache: pip
+      - name: Enforce allowed branches for manual publishing
+        shell: bash
+        run: |
+          validate_branch() {
+            local branch="$1"
+            local label="$2"
+            if [[ "$branch" != "main" && ! "$branch" =~ ^hotfix/.+ ]]; then
+              echo "::error::${label} branch '$branch' is not allowed. Only 'main' or 'hotfix/*' are permitted."
+              exit 1
+            fi
+            echo "${label} branch '$branch' is valid."
+          }
+
+          CURRENT_BRANCH="${{ github.ref_name }}"
+          INPUT_BRANCH="${{ inputs.branch }}"
```

Comment:
This is a duplicate of what [environments](https://github.com/scality/mountpoint-s3-csi-driver/settings/environments) can do for you. Create one or use the `github-pages` existing one. And configure it to your liking.

You can even add a manual approval of a build if you'd like.

Reply:
I removed this workflow as discussed yesterday